Convert Datastore Dashboard Controller into Form

clayliddell opened the pull request:
- [x] Test coverage exists
- [ ] Documentation exists

## QA Steps

- [ ] Create a dataset via a harvest file.
- [ ] Ensure the DKAN instance you are working with has at least 11 datasets.
- [ ] Navigate to the datastore dashboard.
- [ ] Ensure only 10 datasets are showing up on the first page, and other datasets can be accessed via the pager.
- [ ] Ensure the UUID filter is working on the datastore dashboard.
- [ ] Ensure the harvest filter is working on the datastore dashboard.
- [ ] Ensure links for harvest IDs on the harvest dashboard navigate to a pre-filtered datastore dashboard.

Commit:
Convert Dashboard Controller into Form.

Commit:
Cleaning up dashboard form.

Make filters inline and convert harvest ID filter to dropdown.

Commit:
Add test coverage for new dashboard form.

Commit:
Remove accidental changes to ResourceLocalizer

Commit:
Swap unnecessary double-quotes for single-quotes

dafeder commented on modules/datastore/src/Form/DashboardForm.php:
```diff
@@ -0,0 +1,440 @@
+  /**
+   * Fetch the status of all harvests.
+   *
+   * @return \Generator
```
@clayliddell idk if upgrading codeclimate will fix this, but really this is not a very useful piece of documentation anyway - it's technically correct but doesn't really tell you what is being yielded. I'd say just remove this line.

clayliddell replied:
Fair enough. Resolved in https://github.com/GetDKAN/dkan/pull/3714#discussion_r760346155.

dafeder reviewed (changes requested):
Let's override codeclimate on the buildFilters() method -- sometimes Drupal's just gonna Drupal -- but remove that `@return`

clayliddell commented on modules/datastore/src/Form/DashboardForm.php:
```diff
@@ -0,0 +1,440 @@
+
+  /**
+   * Fetch the status of all harvests.
+   *
+   * @return \Generator
+   *   Dataset harvest statuses.
```
```suggestion
```

Commit:
Remove unnecessary @return in DocBlock to satisfy CodeClimate

Prevent CodeClimate from complaining about me using a @return comment on a method with a yield statement.

clayliddell commented on modules/datastore/src/Form/DashboardForm.php:
```diff
@@ -0,0 +1,437 @@
+   *
+   * @return array[]
+   *   Table filters render array.
+   */
```
```suggestion
   *
   * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
   */
```

dafeder replied:
hmm neat trick. ugly but effective!

dafeder replied:
heh nevermind, does not look like CC catches that

clayliddell replied:
Yeah, I've tried everything I can think of. I thought this particular warning was coming from PHPMD, but I guess not...

> Let's override codeclimate on the buildFilters() method -- sometimes Drupal's just gonna Drupal...

How I can do this?

Commit:
Suppress excessive length warning on ::buildFilters()

Commit:
Try disabling PHPCS for ::buildFilters()

Commit:
Swap out PHPCS v3 code standard comments for v2

Commit:
Maybe this will work?

Commit:
Remove unnecessary doc comments
